src/app/api/utils/auth.tsx

```diff
@@ -81,6 +81,31 @@ export const authOptions: AuthOptions = {
   },
   providers: [fxaProviderConfig],
   callbacks: {
+    redirect({ url, baseUrl }) {
+      // Do not pass un-parseable URLs to next-auth, @see MNTOR-3029
+      // Preserve default behavior otherwise. @see https://next-auth.js.org/configuration/callbacks#redirect-callback
+
+      // Allows relative callback URLs
+      if (url.startsWith("/")) {
+        if (!URL.canParse(`${baseUrl}${url}`)) {
+          logger.warn("Cannot parse redirect url:", {
```

Comment:
I half remember Joe mentioning that we had a convention to use snake_cased identifiers for our logs? e.g.

```suggestion
          logger.warn("nextauth_relative_redirect_url_parse_error:", {
```